rasa/nlu/components.py

```diff
@@ -203,6 +211,69 @@ def validate_required_components_from_data(
             "no response selector component specified inside NLU pipeline."
         )
 
+    if data.entity_examples and not any_components_in_pipeline(
+        TRAINABLE_EXTRACTORS, pipeline
+    ):
+        raise_warning(
+            "You have defined training data consisting of entity examples, but "
+            "your NLU pipeline does not include an entity extractor trained on "
+            "your training data. To extract entity examples, add one of "
+            f"{TRAINABLE_EXTRACTORS} to your pipeline."
+        )
+
+    if data.regex_features and not any_components_in_pipeline(
+        ["RegexFeaturizer"], pipeline
+    ):
+        raise_warning(
+            "You have defined training data with regexes, but "
+            "your NLU pipeline does not include a RegexFeaturizer. "
+            "To featurize regexes for entity extraction, you need "
```

Comment:
Minor: The features from the `RegexFeaturizer` are also used by DIET. So not only entity extraction but also intent classification.

Reply:
@tabergma Hm. So hypothetically the checks about whether the `pattern` feature is turned on or not may not be relevant, if they don't do entity extraction at all? E.g. if they have lookup tables, diet with intent only, and no CRF, that's valid?

Reply:
Yes. in that pipeline the pattern features would just be used for intent classification.

Reply:
Got it. so then this check
1. check for regex featurizer (everything needs this right?)
2. check for diet OR CRF
3. If CRF, check for pattern feature
would then erroneously be adding warnings 🤔